### ISSUE TITLE
Create any data structure in any language

### ISSUE THREAD

**Issue by BaReinhard** [BaReinhard/Hacktoberfest-Data-Structure-and-Algorithms#60]
Be sure to follow the structure 
data_structures/general/specific/language/file.lang

Ex 
data_structures/binary_tree/javascript/BST.js

**Comment by EduardoGR** [comment]
Dibs!!

**Comment by BaReinhard** [comment]
No preferences as of now. Whatever you feel like experimenting with.

**Comment by Prohunt** [comment]
Will implement Java Stack

**Comment by srpurwaha201** [comment]
Can i implement linkedList in c++?

**Comment by mhdsmr2** [comment]
Please merge my [PR](https://github.com/BaReinhard/Hacktoberfest-Data-Structure-and-Algorithms/pull/159) which will complete issue #133 and also this one.

**Comment by Kushredoc** [comment]
I want to add a Graph in Java using Hashmap and Linkedlist , how should I proceed .
I wish to create a folder 'Java' under Data Structures/ Graph 
and then add my files .

**Comment by mhdsmr2** [comment]
Here's what you do:
1.) Fork this repo.
2.) Clone your repo on your local machine.
3.) Go to Data Structures/Graph and create a new folder named "Java".
4.) Add your files to that folder.
5.) Push it your remote repo.
6.) There should a 'create a pull request' button, click on it.
7.) Add the title and the description to the PR(and don't forget to mention this issue in the PR. You can mention it by adding 'Does #60' in your description). My suggestion would be to add 'Add Graph implementation in Java' as title and 'Closes issue #60' as the description.
And you're done!
(Ask anything that you didn't understand).

**Comment by Kushredoc** [comment]
I added graph in java PR #275 . 
Do tell me how can I improve it , also I read the issue #133 just now , so didn't add any explanation .

**Comment by mhdsmr2** [comment]
Change the title of your PR to 'Add Graph's implementation in Java' and change the description to 'Added Graph's implementation in Java for issue #60'. I think that'll be enough. :)

**Comment by Kushredoc** [comment]
Done and thanks for helping me out ! :D

**Comment by mhdsmr2** [comment]
Cool. ;)

**Comment by gunanksood** [comment]
hey! i want to add a graph problem in c++. Assign this issue to me @BaReinhard

**Comment by gunanksood** [comment]
i have added Application of DFS algorithm to find largest component in a matrix. please merge the PR #327

**Comment by jasm123** [comment]
Can I add  a graph in Java?

**Comment by mhdsmr2** [comment]
No. @Kushredoc Did that already. You have to use a different language or choose another Data Structure. Sorry.

**Comment by jasm123** [comment]
Okay, then can I add a Segment Tree in Java ? 
Also, in Graph, I've added DFS, BFS and topological sorting in addition.

**Comment by mhdsmr2** [comment]
The folders don't exist as of now so you need to create the folders yourself.
Other than that, it's all great(just remember to create the BFS, DFS and the topological sorting folders in the Graph folder). Add them in different PRs(you may need to create different branches to have a clean commit history and not having the changes overlap).

**Comment by jasm123** [comment]
Yes, sure. Will do them in a day. Thanks!

**Comment by mhdsmr2** [comment]
Welcome! :-)

**Comment by Gupta-shifali** [comment]
Can I add c++ code for doubly linked list @BaReinhard??

**Comment by jasm123** [comment]
Created seperate folder and have submitted PRs for segment tree. 
I didn't submit PR for BFS, DFS because it is already there in algorithms folder.

**Comment by mhdsmr2** [comment]
Cool. +1.

**Comment by IamRaviTejaG** [comment]
Caesar Cipher in C++ & Python is up here in my PR: https://github.com/BaReinhard/Hacktoberfest-Data-Structure-and-Algorithms/pull/439

Please merge the PR.

**Comment by raghunanj** [comment]
I have added files for the union data structure in python with an example algorithm problem(graph). Please merge PR #631

**Comment by m-sameer** [comment]
NOTICE
You don't have to mention that you have made a PR. You may just mention any one of the assignees for this issue in your PR.

**Comment by d4nielhdz** [comment]
#836 **Implemented MinHeap in C++**

**Comment by nishamehra** [comment]
Can I add a detect a loop in linked list in java?

**Comment by Akriti-05** [comment]
May I implement min stack using java?

**Comment by Aarav238** [comment]
i want to implement data structure in java
Please assign this issue to me.
I can implement every data structure in java

**Comment by tripathisharad** [comment]
Hi @BaReinhard I would like contribute on this project can you please assign me.

**Comment by arpanbiki** [comment]
Sir please assign me this issue.
